### PR TITLE
KOTOR: Party Management

### PR DESCRIPTION
**Pull request by Nostritius**
This PR tries to implement a basic version of the party management for kotor

### CODE REVIEW

File: src/engines/kotor/gui/ingame/hud.cpp [hunk 1]
```diff
@@ -148,6 +149,51 @@ void HUD::showContainer() {
 	sub(*_container, kStartCodeNone, true, false);
 }
 
+void HUD::setPartyLeader(Creature *creature) {
+	if (!creature) {
+		getLabel("LBL_BACK1")->setInvisible(true);
+		getLabel("LBL_CHAR1")->setInvisible(true);
+		getProgressbar("PB_VIT1")->setInvisible(true);
```
Comment (DrMcCoy) [inline src/engines/kotor/gui/ingame/hud.cpp]: You forgot a return here

File: src/engines/kotor/gui/ingame/hud.cpp [hunk 2]
```diff
@@ -148,6 +149,51 @@ void HUD::showContainer() {
...
+	getLabel("LBL_CHAR1")->setFill(creature->getPortrait());
+
+	getProgressbar("PB_VIT1")->setInvisible(false);
+}
+
+void HUD::setPartyMember1(Creature *creature) {
+	if (!creature) {
+		getLabel("LBL_BACK2")->setInvisible(true);
+		getLabel("LBL_CHAR2")->setInvisible(true);
+		getProgressbar("PB_VIT2")->setInvisible(true);
```
Comment (DrMcCoy) [inline src/engines/kotor/gui/ingame/hud.cpp]: You forgot a return here

File: src/engines/kotor/gui/ingame/hud.cpp [hunk 3]
```diff
@@ -148,6 +149,51 @@ void HUD::showContainer() {
...
+	getLabel("LBL_CHAR2")->setFill(creature->getPortrait());
+
+	getProgressbar("PB_VIT2")->setInvisible(false);
+}
+
+void HUD::setPartyMember2(Creature *creature) {
+	if (!creature) {
+		getLabel("LBL_BACK3")->setInvisible(true);
+		getLabel("LBL_CHAR3")->setInvisible(true);
+		getProgressbar("PB_VIT3")->setInvisible(true);
```
Comment (DrMcCoy) [inline src/engines/kotor/gui/ingame/hud.cpp]: You forgot a return here

File: src/engines/kotor/module.cpp [hunk 4]
```diff
@@ -701,6 +705,42 @@ void Module::movedPC() {
...
+	_party.push_back(creature);
+
+	_ingame->setPartyLeader(*_party.begin());
+
+	if (_party.size() == 2) {
+		std::list<Creature *>::iterator iter = _party.begin();
+		std::advance(iter, 1);
+		_ingame->setPartyMember1(*iter);
+	} else {
+		_ingame->setPartyMember1(0);
```
Comment (DrMcCoy) [inline src/engines/kotor/module.cpp]: This method is completely wrong and broken, and doesn't do what you want it to do.

- So when adding party member 3, party.size() == 3, so you call setPartyMember1(0)?
- Why are you re-setting the party leader each time?
- Why are you iterating over the list when you already have creature?

File: src/engines/kotor/module.cpp [hunk 5]
```diff
@@ -701,6 +705,42 @@ void Module::movedPC() {
...
+		_ingame->setPartyMember2(*iter);
+	} else {
+		_ingame->setPartyMember2(0);
+	}
+
+	// TODO: If the party size increases over 3 show the character selection screen.
+}
+
+bool Module::isObjectPartyMember(Creature *creature) {
+	for (std::list<Creature *>::iterator iter = _party.begin(); iter != _party.end(); ++iter) {
```
Comment (DrMcCoy) [inline src/engines/kotor/module.cpp]: You could use std::find here instead

File: src/engines/kotor/module.h [hunk 6]
```diff
@@ -225,6 +234,10 @@ class Module : public KotOR::Object, public KotOR::ObjectContainer {
 	std::map<Common::UString, int> _globalNumbers;
 	// '---
 
+	// .--- Party
+	std::list<Creature *> _party;
```
Comment (DrMcCoy) [inline src/engines/kotor/module.h]: That needs to be cleared when the module is unloaded

**Comment by Nostritius** [comment]
The issues should now be fixed

**Comment by DrMcCoy** [comment]
Merged as d79266c8...e231bcd6, thanks! :)

**Pull request closed without merging**